src/ddmd/root/filename.d

```diff
@@ -721,3 +722,124 @@ nothrow:
+
+      Please consult
+      https://msdn.microsoft.com/en-us/library/windows/desktop/aa363855(v=vs.85).aspx
+    */
+    private int _mkdir(const(char)* path) nothrow
+    {
+        import core.sys.windows.winbase: CreateDirectoryW, GetLastError;
+        import core.sys.windows.winerror: ERROR_ALREADY_EXISTS;
+        import core.stdc.errno: errno, EEXIST;
+
+        const extendedPath = path.toExtendedLengthPath();
+
+        // the CreateDirectoryW function takes a pointer to a structure,
+        // but since we're not interested in using it it's declared void*
+        // and we pass null for the default behaviour.
+        DWORD lastError;
+        const createRet = () @trusted
+        {
+            const ret = CreateDirectoryW(&extendedPath[0], null /*securityAttributes*/);
+            lastError = GetLastError();
+            return ret;
+        }();
+
+        // Preserve compatibility with mkdir since the calling code expects
+        // errno to be set.
+        if (createRet == 0 && lastError == ERROR_ALREADY_EXISTS)
+            errno(EEXIST);
+
+        // different conventions for CreateDirectory and mkdir
+        return createRet == 0 ? 1 : 0;
+    }
+
+    // Converts a path to one suitable to be passed to Win32 API
+    // functions that can deal with paths longer than 248
+    // characters. For more information:
+    // https://msdn.microsoft.com/en-us/library/windows/desktop/aa365247(v=vs.85).aspx
+    // buf is used as a scratch space instead of allocating. If its length is insufficient,
+    // memory is allocated on the GC heap.
+    package wchar[] toExtendedLengthPath(const(char*) path, wchar[] buf = []) nothrow
+    {
+        import core.sys.windows.winbase: GetFullPathNameW;
+
+        // Since the unicode Win32 APIs use UTF16, we need to convert
+        // from UTF8 to UTF16 first.
+        const wpath = path.toWStringz(buf);
+
+        // Now wpath is the UTF16 version of path, but to be able to use
+        // extended paths, we need to prefix with `\\?\` and the absolute
+        // path.
+
+        static immutable prefix = `\\?\`w; // see comment above
+
+        wchar[1024] absBuf;
+
+        // We don't know how large the absolute file path will be We
+        // need to tell GetFullPathName the size of the buffer we're
+        // passing in so we defensively allocate 4 times the size of
+        // the string we have currently.
+        const absLength = prefix.length + wpath.length * 4;
+        auto absPath = absLength > absBuf.length ? new wchar[absLength] : absBuf[];
+
+        absPath[0 .. prefix.length] = prefix[];
+
+        const absPathRet = () @trusted
+        {
+            return GetFullPathNameW(&wpath[0],
+                                    absPath.length,
+                                    &absPath[prefix.length],
+                                    null /*filePartBuffer*/);
+        }();
+
+        return absPathRet > absPath.length ? null : absPath[0 .. absPathRet];
```

Comment:
This could be returning a slice of `absBuf`, which is _really_ bad, because it's a local static array. If you want to return a slice of a static array, then you'll need to do something like reuse `buf` somehow.

Reply:
Sigh, I forgot to make it static, good catch. I guess reusing `buf` is a solution.

Reply:
Changed it so it doesn't return the slice now.